src/shogun/CMakeLists.txt

```diff
@@ -351,6 +351,50 @@ IF (NOT HAVE_STD_VARIANT)
   )
 ENDIF()
 
+CHECK_CXX_SOURCE_COMPILES("
+template<typename... Args>
+auto func(Args... args) {
+    return (args + ...);
+}
+int main()
+{return func(1,2,3);}
+" HAVE_FOLDING_EXPRESSIONS)
+
+IF(NOT HAVE_FOLDING_EXPRESSIONS)
+  MESSAGE(FATAL_ERROR "Compiling shogun requires a compiler with folding expression support!")
+ENDIF()
```

Comment:
right now this doesn't have to be fatal, but would be good to be able to rely on this